### PR TITLE
Adding a mapping from OSM version to supported Ingress version

### PR DESCRIPTION
**Pull request by draychev**
This PR adds mapping from OSM Controller version to supported Ingress version (of the given OSM Controller version).

This is **not** used _yet_.

---

Script to automate the creation of this mapping:
```bash
for x in $(seq 5 9); do git checkout "upstream/release-v0.${x}"; echo "\n----\nv0.$x"; cat ./pkg/ingress/client.go | grep 'ingressInterface\.'; done; git checkout main
```

### CODE REVIEW

File: pkg/osm/ingress.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+		"extensions/v1beta1",
+	},
+	"v0.6": {
+		// Source: https://github.com/openservicemesh/osm/blob/release-v0.6/pkg/ingress/client.go#L6
+		"extensions/v1beta1",
+	},
+	"v0.7": {
+		// Source: https://github.com/openservicemesh/osm/blob/release-v0.7/pkg/ingress/client.go#L6
+		"extensions/v1beta1",
+	},
```
Comment (shashankram) [inline pkg/osm/ingress.go:16]: I don't see a reason to maintain releases prior to v0.7. These are not even documented anymore and as such are not supported. Let's remove this.

File: pkg/osm/ingress.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	"v0.8": {
+		// Source: https://github.com/openservicemesh/osm/blob/release-v0.8/pkg/ingress/client.go#L6
+		"networking/v1beta1",
+	},
+	"v0.9": {
+		// Source: https://github.com/openservicemesh/osm/blob/release-v0.9/pkg/ingress/client.go#L6-L7
+		"networking/v1",
+		"networking/v1beta1",
+	},
+}
```
Comment (shashankram) [inline pkg/osm/ingress.go:26]: In a future release, dependency on k8s ingress will be removed. Would you mind adding a TODO or creating a GH issue so we remember to come back to this later.

Comment (draychev) [reply pkg/osm/ingress.go:26]: @shashankram Not sure what you mean "come back to this later".
I don't anticipate we will go back retroactively and remove `networking/v1beta1` from `v0.7` - so this is here to stay.

File: pkg/osm/ingress.go (hunk 1 above)
Comment (draychev) [reply pkg/osm/ingress.go:16]: We do have release-v0.* branches. Perhaps we should clean these up too.
I'm happy to delete some of these items at some point if we find them unnecessary.

Comment (shashankram) [reply pkg/osm/ingress.go:16]: Release branches should not be removed. But we don't need to keep supporting older releases in code, such as this one.

**Review by shashankram — APPROVED** [review]

**Pull request merged**